File: src/types/lan-share.ts
// SPDX-License-Identifier: GPL-3.0-or-later
// License: GNU GPLv3 or later. See the license file in the project root for more information.
// Copyright © 2021 - present Aleksey Hoffman. All rights reserved.

export type LanShareMode = 'stream' | 'ftp';

export type LanSharePendingReplace = {
  originalPath: string;
  mode: LanShareMode;
  isFile: boolean;
  openPopover: boolean;
  hubPaths?: string[];
};

File: src-tauri/src/lan_share.rs
// SPDX-License-Identifier: GPL-3.0-or-later
// License: GNU GPLv3 or later. See the license file in the project root for more information.
// Copyright © 2021 - present Aleksey Hoffman. All rights reserved.

mod commands;
mod handlers;
mod mdns;
mod network;
mod server;
mod streaming;
mod tls;
mod types;

pub use commands::*;

#[allow(unused_imports)]
pub use types::LanShareResult;
